exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/PrometheusHttpServer.java

```diff
@@ -157,7 +166,16 @@ public void close() {
 
   @Override
   public String toString() {
-    return "PrometheusHttpServer{address=" + getAddress() + "}";
+    StringJoiner joiner = new StringJoiner(",", "PrometheusHttpServer{", "}");
+    joiner.add("host=" + host);
+    joiner.add("port=" + port);
+    joiner.add("otelScopeEnabled=" + otelScopeEnabled);
+    joiner.add("allowedResourceAttributesFilter=" + allowedResourceAttributesFilter);
+    joiner.add("memoryMode=" + memoryMode);
+    joiner.add(
+        "defaultAggregationSelector="
+            + DefaultAggregationSelector.asString(defaultAggregationSelector));
+    return joiner.toString();
```

Comment:
This PR also improves the `PrometheusHttpServer#toString()` implementation, which has been failing to include the various configurable parameters.